db/migrate/20180604143945_create_books.rb

```diff
@@ -0,0 +1,14 @@
+class CreateBooks < ActiveRecord::Migration[5.1]
+  def change
+    create_table :books do |t|
+      t.string :genre
+      t.string :author
+      t.string :image
+      t.string :title
+      t.string :publisher
+      t.string :year
```

Comment:
Here you can add the parameter `null: false`, to the required parameters. That way you have the validation at a database level also.